### PR TITLE
Convert IASI L2 reader for xarray/dask

### PR DESCRIPTION
**Pull request by pnuu**
This PR converts the IASI L2 reader to be compatible with xarray/dask version of SatPy.

This is one part of #399 .

 - [x] Tests added 
 - [x] Tests passed 
 - [x] Passes ``git diff origin/master **/*py | flake8 --diff``

### CODE REVIEW

File: satpy/readers/iasi_l2.py [hunk 1]
```diff
@@ -105,62 +113,69 @@ def get_dataset(self, key, info):
...
     dsid = DSET_NAMES[key.name]
-    data = fid["/PWLR/" + dsid].value
+    dset = fid["/PWLR/" + dsid]
+    if dset.ndim == 3:
+        dims = ['y', 'x', 'level']
+    else:
+        dims = ['y', 'x']
+    data = xr.DataArray(da.from_array(dset.value, chunks=CHUNK_SIZE),
+                        name=key.name, dims=dims).astype(np.float32)
+    data = xr.where(data > 1e30, np.nan, data)
```
Comment (djhoese) [inline satpy/readers/iasi_l2.py]: I don't know if there is a performance difference, but FYI there is also a `data.where(data <= 1e30)` which will fill anything with NaN where the condition is False. But what you have is probably equivalent under the hood.

File: satpy/readers/iasi_l2.py [hunk 2]
```diff
@@ -105,62 +113,69 @@ def get_dataset(self, key, info):
...
-    each of IASI scans."""
+    """Calculate seconds since EPOCH from days and milliseconds for each of IASI scan."""
 
     all_datetimes = []
     for i in range(days.size):
         day = int(days[i])
         msec = msecs[i]
         scanline_datetimes = []
-        for j in range(VALUES_PER_SCAN_LINE / 4):
+        for j in range(int(VALUES_PER_SCAN_LINE / 4)):
```
Comment (djhoese) [inline satpy/readers/iasi_l2.py]: Is this ever going to be not a whole number? I suppose the `int` ends up flooring it anyway.

File: satpy/readers/iasi_l2.py [hunk 3]
```diff
@@ -105,62 +113,69 @@ def get_dataset(self, key, info):
...
             usec = 1000 * (j * VIEW_TIME_ADJUSTMENT + msec)
             for k in range(4):
                 delta = (dt.timedelta(days=day, microseconds=usec))
-                scanline_datetimes.append(EPOCH + delta)
+                scanline_datetimes.append(delta.total_seconds())
```
Comment (djhoese) [inline satpy/readers/iasi_l2.py]: This results in a different result than what this function was doing previously, right? Is this going to confuse/affect what users see?

**Comment by djhoese** [comment]
How hard would it be to add tests for this reader?

Coverage...coverage...coverage!

File: satpy/readers/iasi_l2.py (hunk 1 above)
Comment (pnuu) [reply satpy/readers/iasi_l2.py]: Good to know, although I prefer to have the chosen data shown explicitly. At least for this is more understandable.

File: satpy/readers/iasi_l2.py (hunk 2 above)
Comment (pnuu) [reply satpy/readers/iasi_l2.py]: I added the call to `int()` as `range()` doesn't allow the argument to be a float, which it is for every (single-`/`) division in Python 3. For Python 2 the call isn't needed.

File: satpy/readers/iasi_l2.py (hunk 3 above)
Comment (pnuu) [reply satpy/readers/iasi_l2.py]: Pyresample doesn't want to have the datetime objects in arrays, so this was more or less a must. Maybe this was handled differently in the non-xarray SatPy. And I'm quite certain I've been the only user of this so far, so we shouldn't get any complaints 8-)

**Comment by pnuu** [comment]
Tests should be quite easy, just need to create an dummy file and use that. The format is simple. I'll add something, maybe even today.

**Comment by pnuu** [comment]
I added tests, but just realized I'm using `Scene()` to test the reader. Should I use another day to convert and test directly the reader class?

**Comment by djhoese** [comment]
It is probably more unit-y if you used the Reader directly, but it is not easy to use unless you look at some of the other reader tests. I'm fine either way.

**Comment by pnuu** [comment]
Coverage, coverage, coverage added: 70+ % total.

**Pull request merged**